nifi-extension-bundles/nifi-aws-bundle/nifi-aws-processors/src/main/java/org/apache/nifi/processors/aws/s3/GetS3ObjectMetadata.java

```diff
@@ -0,0 +1,192 @@
+            .build();
+
+    public static Relationship REL_NOT_FOUND = new Relationship.Builder()
+            .name("not found")
+            .description("No object was found in the bucket the supplied key")
+            .build();
+
+    @Override
+    public Set<Relationship> getRelationships() {
+        return new HashSet<>(List.of(REL_FOUND, REL_NOT_FOUND, REL_FAILURE));
+    }
+
+    @Override
+    protected List<PropertyDescriptor> getSupportedPropertyDescriptors() {
+        return properties;
+    }
+
+    @Override
+    public void onTrigger(ProcessContext context, ProcessSession session) throws ProcessException {
+        FlowFile flowFile = session.get();
+        if (flowFile == null) {
+            return;
+        }
+
+        boolean isRouter = context.getProperty(MODE).getValue().equals(MODE_ROUTER.getValue());
+
+        final AmazonS3Client s3;
+        try {
+            s3 = getS3Client(context, flowFile.getAttributes());
+        } catch (Exception e) {
+            getLogger().error("Failed to initialize S3 client", e);
+            flowFile = session.penalize(flowFile);
+            session.transfer(flowFile, REL_FAILURE);
+            return;
+        }
+
+        final String bucket = context.getProperty(BUCKET_WITH_DEFAULT_VALUE).evaluateAttributeExpressions(flowFile).getValue();
+        final String key = context.getProperty(KEY).evaluateAttributeExpressions(flowFile).getValue();
+
+        try {
+            Relationship route;
+
+            try {
+                ObjectMetadata metadata = s3.getObjectMetadata(bucket, key);
+                Map<String, Object> combinedMetadata = new HashMap<>(metadata.getRawMetadata());
+                combinedMetadata.putAll(metadata.getUserMetadata());
+
+                String metadataJson = MAPPER.writeValueAsString(combinedMetadata);
+
+                if (!isRouter && context.getProperty(METADATA_TARGET).getValue().equals(TARGET_ATTRIBUTE.getValue())) {
+                    String attribute = context.getProperty(METADATA_ATTRIBUTE).getValue();
+                    flowFile = session.putAttribute(flowFile, attribute, metadataJson);
+                } else if (!isRouter && context.getProperty(METADATA_TARGET).getValue().equals(TARGET_FLOWFILE_BODY.getValue())) {
+                    flowFile = session.write(flowFile, os -> os.write(metadataJson.getBytes(StandardCharsets.UTF_8)));
+                }
+
+                route = REL_FOUND;
+            } catch (AmazonS3Exception e) {
+                if (e.getStatusCode() == 404) {
+                    route = REL_NOT_FOUND;
+                    flowFile = extractExceptionDetails(e, session, flowFile);
+                } else {
+                    throw e;
+                }
+            }
+
+            session.transfer(flowFile, route);
+        } catch (IOException | AmazonClientException ex) {
+            getLogger().error("There was a problem checking for " + String.format("s3://%s%s", bucket, key), ex);
```

Comment:
Log statements should use placeholders instead of concatenation:
```suggestion
            getLogger().error("Failed to get S3 Object Metadata from s3://{}{} ", bucket, key, e);
```